src/test/java/io/aiven/kafka/connect/common/output/WithMetadataJsonLinesOutputWriterTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.config.OutputFieldEncodingType;
+import io.aiven.kafka.connect.common.config.OutputFieldType;
+import io.aiven.kafka.connect.common.output.jsonwriter.JsonLinesOutputWriter;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+class WithMetadataJsonLinesOutputWriterTest extends JsonLinesOutputWriterTest {
+    private final OutputFieldEncodingType noEncoding = OutputFieldEncodingType.NONE;
+
+    @BeforeEach
+    void setUp() {
+        byteStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void jsonValueWithoutMetadataAndValue() {
+        final List<OutputField> fields = Arrays.asList();
+        sut = JsonLinesOutputWriter.builder().addFields(fields).build();
+        final Schema level1Schema = SchemaBuilder.struct().field("name", Schema.STRING_SCHEMA);
+        final Struct struct1 = new Struct(level1Schema).put("name", "John");
+
+        final SinkRecord record1 = createRecord("key0", level1Schema, struct1, 1, 1000L);
+
+        final String expected = "{}";
+
+        assertRecords(Arrays.asList(record1), expected);
```

Comment:
FWIW, `Arrays.asList` could be replaced with `Collections.singletonList` for single element lists.

Reply:
- [ ] Use SingletonList